itest/rust/src/rect2i_test.rs

```diff
@@ -0,0 +1,106 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::itest;
+use godot::prelude::{inner::InnerRect2i, *};
+
+macro_rules! equiv_mapping {
+    ($fn:ident, $selfarg:expr) => {
+        let outer_res = Rect2i::$fn($selfarg);
+        let inner_res = InnerRect2i::$fn(&InnerRect2i::from_outer(&$selfarg));
+        assert_eq!(outer_res, inner_res);
+    };
+    ($fn:ident -> $t:ty, $selfarg:expr) => {
+        let outer_res = Rect2i::$fn($selfarg);
+        let inner_res = InnerRect2i::$fn(&InnerRect2i::from_outer(&$selfarg)) as $t;
+        assert_eq!(outer_res, inner_res);
+    };
+    ($fn:ident, $selfarg:expr, $($args:expr),*) => {
+        let outer_res = Rect2i::$fn($selfarg, $($args),*);
+        let inner_res = InnerRect2i::$fn(&InnerRect2i::from_outer(&$selfarg), $($args),*);
+        assert_eq!(outer_res, inner_res);
+    };
+}
+
+macro_rules! equiv_mapping_inner_typechange {
+    ($fn:ident, $selfarg:expr, $($args:expr, $t:ty),*) => {
```

Comment:
Maybe the repeating expression could be `$($args:expr : $t:ty)` (colon instead of comma).

This would allow to write expr-type pairs on the same line and make it more readable on call site.

Reply:
![Screenshot_2023-04-07_20-13-30](https://user-images.githubusercontent.com/130095760/230657584-d43de346-b5ac-4458-849a-eaab43a6d823.png)
Sadly the parser isn't happy about this. What do you say about keeping `,`, but parenthesizing the arg-type pairs?

i.e. `equiv_mapping_inner_typechange!(foo, bar, (x, i64), ...)`

Reply:
It looks like you don't actually need `:expr`, but `:ident` should be enough, no?

Reply:
The uppercase is something reserved for :ty, or are there other designators which should be uppercase?

Reply:
It's just types for now. I originally used `$snake_case` everywhere, but found that it significantly improved readability to treat `$Types` like they are in normal Rust code.

I'll put it on my list of code style to document 🙂